src/actions.ts

```diff
@@ -207,12 +207,12 @@ export function resolveSend<TContext, TEvent extends EventObject>(
       : action.event
   );
 
-  let resolvedDelay: number | string | undefined;
+  let resolvedDelay: number | undefined;
   if (isString(action.delay)) {
     const configDelay = delaysMap && delaysMap[action.delay];
     resolvedDelay = isFunction(configDelay)
       ? configDelay(ctx, _event.data, meta)
-      : configDelay || action.delay;
+      : configDelay;
```

Comment:
this change is causing this test to fail - https://github.com/davidkpiano/xstate/blob/5f784e1d358d00ecfd4cc6a6f7ade771f4c05b64/test/after.test.ts#L160

I've removed it as something to discuss - not sure if that's really needed. We warn about unresolved delay strings just after resolveSend gets called and if we were not able to resolve something it seems conceptually better to leave that as undefined - which makes the output SendActionObject a "correct" one, instead of preserving the string and having to deal with such possibility later on.

Reply:
Yeah, that's true, we should remove this test or make it so that it returns `undefined` as the delay (which should be a `number | undefined`).

Reply:
Changed the test, should be ready to merge now